src/connector/src/sink/postgres.rs

```diff
@@ -0,0 +1,625 @@
+                                .execute_raw(self.merge_statement.as_ref().unwrap(), row.iter())
+                                .await?;
+                        }
+                        Op::Delete => {
+                            self.client
+                                .execute_raw(
+                                    self.delete_statement.as_ref().unwrap(),
+                                    row.project(&self.pk_indices).iter(),
+                                )
+                                .await?;
+                        }
+                        Op::UpdateDelete => {}
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl SinkWriter for PostgresSinkWriter {
+    async fn begin_epoch(&mut self, _epoch: u64) -> Result<()> {
+        Ok(())
+    }
+
+    async fn write_batch(&mut self, chunk: StreamChunk) -> Result<()> {
+        let cardinality = self.buffer.push(chunk);
+        if cardinality >= self.config.max_batch_rows {
+            self.flush().await?;
+        }
+        Ok(())
+    }
+
+    async fn barrier(&mut self, is_checkpoint: bool) -> Result<Self::CommitMetadata> {
+        if is_checkpoint {
+            self.flush().await?;
+        }
+        Ok(())
+    }
+
+    async fn abort(&mut self) -> Result<()> {
+        Ok(())
+    }
+
+    async fn update_vnode_bitmap(&mut self, _vnode_bitmap: Arc<Bitmap>) -> Result<()> {
+        Ok(())
+    }
+}
+
+fn data_type_not_supported(data_type_name: &str) -> SinkError {
+    SinkError::Postgres(anyhow!(format!(
+        "{data_type_name} is not supported in SQL Server"
+    )))
+}
+
+fn check_data_type_compatibility(data_type: DataType) -> Result<()> {
+    match data_type {
+        DataType::Boolean
+        | DataType::Int16
+        | DataType::Int32
+        | DataType::Int64
+        | DataType::Float32
+        | DataType::Float64
+        | DataType::Decimal
+        | DataType::Date
+        | DataType::Varchar
+        | DataType::Time
+        | DataType::Timestamp
+        | DataType::Timestamptz
+        | DataType::Jsonb
+        | DataType::Interval
+        | DataType::Bytea => Ok(()),
+        DataType::Struct(_) => Err(data_type_not_supported("Struct")),
+        DataType::List(_) => Err(data_type_not_supported("List")),
+        DataType::Serial => Err(data_type_not_supported("Serial")),
+        DataType::Int256 => Err(data_type_not_supported("Int256")),
+        DataType::Map(_) => Err(data_type_not_supported("Map")),
+    }
+}
+
+fn create_insert_sql(schema: &Schema, table_name: &str) -> String {
+    let columns: String = schema
+        .fields()
+        .iter()
+        .map(|field| field.name.clone())
+        .collect_vec()
+        .join(", ");
+    let parameters: String = (0..schema.fields().len())
+        .map(|i| format!("${}", i + 1))
+        .collect_vec()
+        .join(", ");
+    format!("INSERT INTO {table_name} ({columns}) VALUES ({parameters})")
```

Comment:
Will manually assembling SQL lead to SQL injection vulnerabilities? I would suggest adopting prepared statements.

Reply:
The parameters are bound you may checkout the unit tests.
But I'm not sure if `table_name`, `columns` can be bound. These are symbols, rather than sql values.

Reply:
Ah you're right. It's `$1`.